### ISSUE TITLE
Mish activation function

### ISSUE THREAD

**Issue by nestordemeure** [google/flax#584]
Would you be interested in a PR adding the [Mish activation function](https://github.com/digantamisra98/Mish)?

It is relatively simple but, at least on my use-cases, it does deliver on its promises.

**Comment by sooheon** [comment]
Seems like it'd just be a couple lines, I'm curious what your impl. is? I know there are versions for pytorch with custom cuda kernels, I'd be interested to see how jax perf measures up with just `jit`.

**Comment by nestordemeure** [comment]
The implementation is extremely short indeed:

```python
def mish(x):
    return x * jnp.tanh(softplus(x))
```

It is slower than using a `relu` (which is expected) but I have not benchmarked it against Jitted Pytorch versions to see how Jax does (I would expect it to be competitive).

**Comment by avital** [comment]
Hi @sooheon and @nestordemeure -- as this is not a commonly used non-linearity, and it's a single line implementation, I am not sure it makes sense to put it in Flax core. (Flax doesn't yet have much more common things like common loss functions). 

If a correct and efficient implementation requires tricky custom code (like a custom gradient, or something special to deal with numerical stability) then that makes more sense for us to support, so that others can follow trustworthy patterns and not shoot themselves in the foot. And in that case, probably makes more sense for this to live in JAX core, alongside other activation functions (such as https://jax.readthedocs.io/en/latest/_modules/jax/_src/nn/functions.html#relu). The Flax activation functions are just imported from JAX and re-exported.

I'll close this ticket, but if people feel that this is the wrong call please chime in!
